test/hamiltonian.jl

```diff
@@ -7,3 +8,9 @@ h = Hamiltonian(UnitEuclideanMetric(D), logπ, ∂logπ∂θ)
 r_init = ones(D)
 
 @test AdvancedHMC.kinetic_energy(h, r_init, θ_init) == D / 2
+
+z1 = PhasePoint([NaN], [NaN], DualValue(0.,[0.]), DualValue(0.,[0.]))
+z2 = PhasePoint([Inf], [Inf], DualValue(0.,[0.]), DualValue(0.,[0.]))
+
+@test z1.logπ.value == z1.logπ.value
+@test z1.logκ.value == z1.logκ.value
```

Comment:
We could add some tests for constructions that would throw errors?

Reply:
These test cases already test `NaN` and `Inf` values for Kinect and potential energy. `PhasePoint` constructors will throw a warning by default; then we have to detect and handle errors explicitly by calling `isfinite(z::PhasePoint)`.

Reply:
I simply mean testing these warnings explicitly as they are quite important. I added it in https://github.com/TuringLang/AdvancedHMC.jl/pull/51/commits/0d4aed334c70627ce4f5ec8dc5da9c26691c341a